### PR TITLE
Upgrade driver for the RS SGS100A

### PR DESCRIPTION
**Pull request by nulinspiratie**
Adds many additional parameters etc to the RS SGS100A.
Should be good to go, but first PR #52  should be accepted

### CODE REVIEW

**Comment by maij** [comment]
Review blocked, pending #52

File: qcodes/instrument_drivers/rohde_schwarz/SGS100A.py [hunk 1]
```diff
@@ -27,110 +36,220 @@ class RohdeSchwarz_SGS100A(VisaInstrument):
...
+            set_cmd="SOUR:PHAS {:.2f}",
+            get_parser=float,
+            vals=vals.Numbers(0, 360)
+        )
+        self.add_parameter(
+            name="power",
+            label="Power",
+            unit="dBm",
+            get_cmd="SOUR:POW?",
+            set_cmd="SOUR:POW {:.2f}",
```
Comment (maij) [inline qcodes/instrument_drivers/rohde_schwarz/SGS100A.py]: From the [manual](https://www.rohde-schwarz.com/webhelp/sgs100a_html_usermanual_en/SGS100A_HTML_UserManual_en.htm) it looks like the right command is:
`':SOURce:POWer:POWer {:.2f}'`

Does it work without the second power?

File: qcodes/instrument_drivers/rohde_schwarz/SGS100A.py [hunk 2]
```diff
@@ -27,110 +36,220 @@ class RohdeSchwarz_SGS100A(VisaInstrument):
...
+        self.add_parameter(
+            name='maximum_power',
+            label='Maximum power',
+            unit='dBm',
+            get_cmd=lambda: self.power.vals._max_value,
+            set_cmd=lambda power: setattr(self.power.vals, '_max_value', power),
+            initial_value=self.power.vals._max_value,
+            # VISA commands do not seem to influence max power
+            # get_cmd='SOURce:POWer:LIMit?',
+            # set_cmd='SOURce:POWer:LIMit:AMPLitude {:.2f}',
```
Comment (maij) [inline qcodes/instrument_drivers/rohde_schwarz/SGS100A.py]: The manual says:

> Sets the level limit.
> 
> The value specifies the upper limit of the level at the [RF Out] connector. A message appears if an attempt is made to set a level above this limit and the level at the RF output is confined to the upper limit. However, the **level indication is not influenced**.

So maybe it's working and raising an error you're not detecting.

File: qcodes/instrument_drivers/rohde_schwarz/SGS100A.py (hunk 1 above)
Comment (nulinspiratie) [reply qcodes/instrument_drivers/rohde_schwarz/SGS100A.py]: Haha interesting, it seems to work, I'll check

Comment (nulinspiratie) [reply qcodes/instrument_drivers/rohde_schwarz/SGS100A.py]: Yes apparently it works

File: qcodes/instrument_drivers/rohde_schwarz/SGS100A.py (hunk 2 above)
Comment (nulinspiratie) [reply qcodes/instrument_drivers/rohde_schwarz/SGS100A.py]: Good catch! I was wondering why the Rabi frequencies weren't increasing past 0 dBm, forgot I set it to 0 dBm a long long time ago. You get a medal🥇

**Comment by nulinspiratie** [comment]
@maij I think it's good to go

File: qcodes/instrument_drivers/rohde_schwarz/SGS100A.py [hunk 3]
```diff
@@ -27,110 +36,221 @@ class RohdeSchwarz_SGS100A(VisaInstrument):
...
+        )
+        self.add_parameter(
+            name="power",
+            label="Power",
+            unit="dBm",
+            get_cmd="SOUR:POW?",
+            set_cmd="SOUR:POW {:.2f}",
+            get_parser=float,
+            vals=vals.Numbers(-120, 25),
+            docstring="RF output power. Note that if this exceeds max_power, "
```
Comment (maij) [inline qcodes/instrument_drivers/rohde_schwarz/SGS100A.py]:
```suggestion
            docstring="RF output power. Note that if this exceeds maximum_power, " 
```

**Pull request merged**